capture/genBitmapsWebdriver.js

```diff
@@ -0,0 +1,167 @@
+"use strict";
+
+/* eslint-disable no-path-concat */
+const DOCUMENT_SELECTOR = 'document';
+const fs = require('fs');
+const path = require('path');
+const cwd = fs.workingDirectory;
+const exec = require('child_process').exec;
+
+const args = process.argv;
+
+var genConfigPath = args[2];
+
+const config = require(genConfigPath);
+if (!config.paths) {
+    config.paths = {};
+}
+
+console.log("config", config)
+
+const isReference = config.isReference;
+if (isReference) {
+    console.log('CREATING NEW REFERENCE FILES');
+}
+
+var outputFormat = "." + (config.outputFormat && config.outputFormat.match(/jpg|jpeg/) || 'png');
+var bitmapsReferencePath = config.paths.bitmaps_reference || 'bitmaps_reference';
+var bitmapsTestPath = config.paths.bitmaps_test || 'bitmaps_test';
+var casperScriptsPath = config.paths.casper_scripts || null;
+var comparePairsFileName = config.paths.tempCompareConfigFileName;
+// var viewports = config.viewports;
+const scenarios = config.scenarios || config.grabConfigs;
+const configId = config.id || genHash(config.backstopConfigFileName);
+var fileNameTemplate = config.fileNameTemplate || '{configId}_{scenarioLabel}_{selectorIndex}_{selectorLabel}_{viewportIndex}_{viewportLabel}';
+
+const compareConfig = {testPairs: []};
+
+const browser = require("webdriverio");
+const client = browser.remote({
+    browserName: 'chrome'
+});
+
+var wdioElementScreenshot = require('wdio-element-screenshot');
+wdioElementScreenshot.init(client);
+
+client.init()
+  .url(scenarios[0].url)
+  .then(capturePageSelectors);
+
+function capturePageSelectors() {
+    var screenshotNow = new Date();
+    var screenshotDateTime = screenshotNow.getFullYear() + pad(screenshotNow.getMonth() + 1) + pad(screenshotNow.getDate()) + '-' + pad(screenshotNow.getHours()) + pad(screenshotNow.getMinutes()) + pad(screenshotNow.getSeconds());
+    
+    setTimeout(function () {
+        config.scenarios.forEach(function (scenario, i) {
+            var scenarioLabelSafe = makeSafe(scenario.label);
+            scenario.sIndex = i;
+            
+            processScenario(scenario, scenarioLabelSafe, screenshotDateTime);
+        });
+    }, 1000);
+}
+
+function processScenario(scenario, scenarioLabel, screenshotDateTime) {
+    config.viewports.forEach(function (vp, viewportIndex) {
+        
+        console.log('Capturing screenshots for ' + makeSafe(vp.name) + ' (' + (vp.width || vp.viewport.width) + 'x' + (vp.height || vp.viewport.height) + ')', 'info');
+        
+        let l = scenario.selectors.length;
+        let counter = 0;
+        
+        scenario.selectors.forEach(function (selector, i) {
+            let cleanedSelectorName = selector.replace(/[^a-z0-9_\-]/gi, ''); // remove anything that's not a letter or a number
+            
```

Comment:
Functions like these should probably be broken out into separate util files with tests to make this refactor cleaner.